src/pages/auth/signup/signup.jsx

```diff
@@ -1,63 +1,119 @@
+  const [password, setPassword] = useState("");
+const navigate = useNavigate()
+const signupHandler = async () => {
 
-     <div className="signup-form col-5">
-         <h2>Register.</h2>
-         <p>Fill your details here</p>
+const body = {
+  email: email,
+  password: password,
+  firstName: firstName,
+  lastName: lastName,
+}
+ 
+try {
+ const response = await axios.post("/api/auth/signup", body)
+ localStorage.setItem("token", response.data.encodedToken);
+ setTimeout(() => {
+  navigate("/login")
+ }, 1000);
+} catch (error) {
+  console.error(Error)
+}
+}
 
-    <form>
-    <div class="grid-container">
 
-    <div class="grid-item">
-     <label htmlFor="user-first-name">First Name</label>
-    <input id="user-first-name" type="text" />
-    </div>
+  return (
+    <>
+      <div
+        className="outer-box"
+        style={{ backgroundImage: "url(" + signup1 + ")" }}
+      >
+        <div className="signup-form col-5">
+          <h2>Register.</h2>
+          <p>Fill your details here</p>
+
+ 
+            <div class="grid-container">
+              <div class="grid-item">
+                <label htmlFor="user-first-name">First Name</label>
+                <input
+                  id="user-first-name"
+                  type="text"
+                  onChange={(e) => {
+                    e.preventDefault()
+                    setFirstName(e.target.value)}}
+                />
+              </div>
 
-  <div class="grid-item">
-  <label htmlFor="user-last-name">Last Name</label>
-   <input id="user-last-name" type="text" />
-  </div>
+              <div class="grid-item">
+                <label htmlFor="user-last-name">Last Name</label>
+                <input
+                  id="user-last-name"
+                  type="text"
+                  onChange={(e) => {
+                    e.preventDefault()
+                    setLastName(e.target.value)}}
+                />
+              </div>
 
-  <div class="grid-item">
-  <label htmlFor="user-signup-email">Email</label>
-  <input id="user-signup-email" type="email" />
-  </div>  
+              <div class="grid-item">
+                <label htmlFor="user-signup-email">Email</label>
+                <input
+                  id="user-signup-email"
+                  type="email"
+                  onChange={(e) => {
+                    e.preventDefault()
+                    setEmail(e.target.value)}}
+                />
+              </div>
 
-  <div class="grid-item">
+              {/* <div class="grid-item">
   <label htmlFor="user-number">Contact Number</label>
   <input id="user-number" type="number" />
-  </div>
+  </div> */}
 
-  <div class="grid-item">
-  <label htmlFor="user-pass">Create Password</label>
-  <input id="user-pass" type="password" />
-  </div>
+              <div class="grid-item">
+                <label htmlFor="user-pass">Create Password</label>
+                <input
+                  id="user-pass"
```

Comment:
The code is well implemented and use of useState has been done nicely and all modules are used perfectly just a small syggestion regarfing indentation of the code.